packages/admin/cms-admin/src/dam/Table/fileUpload/FileUploadContext.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+
+interface FileUploadContextApi {
+    newlyUploadedFileIds: string[];
+    addNewlyUploadedFileIds: (newlyUploadedFileIds: string[]) => void;
+}
+
+const FileUploadContext = React.createContext<FileUploadContextApi>({
+    newlyUploadedFileIds: [],
+    addNewlyUploadedFileIds: () => {
+        console.warn("If you want to track newlyUploadedFileIds, FileUploadContextProvider has to be defined higher up in the tree");
+    },
+});
+
+export const useFileUploadContext = () => React.useContext(FileUploadContext);
+
+export const FileUploadContextProvider: React.FunctionComponent = ({ children }) => {
+    const [newlyUploadedFileIds, setNewlyUploadedFileIds] = React.useState<string[]>([]);
+
+    const addNewlyUploadedFileIds = (fileIds: string[]) => {
+        setNewlyUploadedFileIds((newlyUploadedFileIds) => [...newlyUploadedFileIds, ...fileIds]);
+
+        setTimeout(() => {
```

Comment:
This may lead to state updates on unmounted components.

Reply:
👍 I resolved this issue